charts/k8s-monitoring/charts/feature-pod-logs/templates/_common_pod_discovery.alloy.tpl

```diff
@@ -61,6 +61,28 @@ discovery.relabel "filtered_pods" {
     regex = "__meta_kubernetes_pod_annotation_(.+)"
   }
 
+  // explicitly set service_name. if not set, loki will automatically try to populate a default.
+  // see https://grafana.com/docs/loki/latest/get-started/labels/#default-labels-for-all-users
+  //
+  // choose the first value found from the following ordered list:
+  // - pod.annotation[resource.opentelemetry.io/service.name]
+  // - pod.label[app.kubernetes.io/name]
+  // - k8s.pod.name
+  // - k8s.container.name
+  rule {
+    action = "replace"
+    source_labels = [
+      "__meta_kubernetes_pod_annotation_resource_opentelemetry_io_service_name",
+      "__meta_kubernetes_pod_label_app_kubernetes_io_name",
+      "__meta_kubernetes_pod_name",
+      "__meta_kubernetes_pod_container_name",
+    ]
+    separator = ";"
+    regex = "^(?:;*)?([^;]+).*$"
+    replacement = "$1"
+    target_label = "service_name"
+  }
```

Comment:
@cyrille-leclerc do you mind taking a look at this?

I can't include other values used by the opentelemetry operator (statefulset, deployment, etc.) because they are not currently supported by prometheus kubernetes service discovery.

Reply:
Looks good to me thanks! I did a quick test on https://regex101.com/ and it seems to work but I didn't test in real life.